### PR TITLE
Fix OptionIdOps.some to always return Some

### LINKED ISSUE #871: OptionIdOps.some inconsistent
While obviously `null` shouldn't be used, if it were to creep in then `x.some` violates the principle of least surprise. Given this code:

```scala
val s: String = null
val o = s.some
```

I might expect `o` to be `Some(...)`, but it ends up as `None`.

### PR DESCRIPTION
**Pull request by vpavkin**
Fix #871

### CODE REVIEW

File: tests/src/test/scala/cats/tests/OptionTests.scala [hunk 1]
```diff
@@ -76,4 +76,11 @@ class OptionTests extends CatsSuite {
       isEq.lhs should === (isEq.rhs)
     }
   }
+
+  // a test for OptionIdOps.some to always return Some
+
+  test("OptionIdOps.some"){
+    val s: String = null
+    s.some.contains(s) should === (true)
```
Comment (vpavkin) [inline tests/src/test/scala/cats/tests/OptionTests.scala]: Due to NPE I wasn't able to write it as `s.some should === (Some(null))`

Comment (ceedubs) [inline tests/src/test/scala/cats/tests/OptionTests.scala]: Thanks, @vpavkin!

Would you mind leaving a comment in the code itself about why it's done this way?

Also, would you mind naming the test something that better identifies that it's checking for behavior when `null` is passed in? Maybe something like `test(".some with null argument still results in Some #871")`.

Comment (ceedubs) [inline tests/src/test/scala/cats/tests/OptionTests.scala]: Looks like `Option.contains` isn't available on Scala 2.10

**Comment by vpavkin** [comment]
@ceedubs my bad. Added more comments and changed code to use `exists`, which should work for 2.10.
Let's see what travis thinks about it.

File: tests/src/test/scala/cats/tests/OptionTests.scala [hunk 2]
```diff
@@ -76,4 +76,12 @@ class OptionTests extends CatsSuite {
       isEq.lhs should === (isEq.rhs)
     }
   }
+
+  // a test for OptionIdOps.some to return Some even if the argument is null
```
Comment (ceedubs) [inline tests/src/test/scala/cats/tests/OptionTests.scala]: This comment probably isn't really needed since the test name says basically the same thing.

**Comment by ceedubs** [comment]
:+1: thanks a bunch, @vpavkin!

**Comment by vpavkin** [comment]
updated comment :)

**Comment by stew** [comment]
:+1:
